regcore/layer.py

```diff
@@ -0,0 +1,22 @@
+class LayerParams(object):
```

Comment:
* This may come down to personal preference, but why is this a class and not a function? I'm going to cite the old standby here: https://www.youtube.com/watch?v=o9pEzgHorH0.
* Can we flag this for deprecation in a future release?